iceoryx_hoofs/include/iceoryx_hoofs/design_pattern/polymorphic_handler.hpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_DESIGN_PATTERN_POLYMORPHIC_HANDLER_HPP
+#define IOX_HOOFS_DESIGN_PATTERN_POLYMORPHIC_HANDLER_HPP
+
+#include <atomic>
+#include <iostream>
+#include <type_traits>
+
+#include "iceoryx_hoofs/design_pattern/static_lifetime_guard.hpp"
+
+namespace iox
+{
+namespace design_pattern
+{
+/// @brief Implements the Activatable concept to be used in the PolymorphicHandler
+/// The concept implements a binary switch. By default is switched on (active).
+/// Anyone defining another custom handler interface is supposed to derive from Activatable.
+/// @note While this is public, it is also partially implementation detail and partially convenience
+/// to use the PolymorphicHandler.
+class Activatable
+{
+  public:
+    Activatable() = default;
+
+    /// @brief Switch on.
+    void activate() noexcept;
+
+    /// @brief Switch off.
+    void deactivate() noexcept;
+
+    /// @brief Query switch state.
+    /// @return true if active (on), false otherwise (off).
+    bool isActive() const noexcept;
+
+  private:
+    bool m_active{true};
+};
+
+namespace detail
+{
+
+///@brief default hooks for the PolymorphicHandler
+///@note template hooks to avoid forced virtual inheritance (STL approach),
+template <typename Interface>
+struct DefaultHooks
+{
+    /// @brief called after if the polymorphic handler is set or reset after finalize
```

Comment:
@FerdinandSpitzschnueffler This is BTW. to allow configuration options. Assuming it is used for a logger, it could log a warning to both, old and new logger, and terminate.